modules/questdb/src/main/java/org/testcontainers/containers/QuestDBContainer.java

```diff
@@ -0,0 +1,85 @@
+package org.testcontainers.containers;
+
+import lombok.NonNull;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Testcontainers implementation for QuestDB.
+ */
+public class QuestDBContainer extends JdbcDatabaseContainer<QuestDBContainer> {
+
+    static final String DATABASE_PROVIDER = "postgresql";
+
+    private static final String DEFAULT_DATABASE_NAME = "qdb";
+
+    private static final int DEFAULT_COMMIT_LAG_MS = 1000;
+
+    private static final String DEFAULT_USERNAME = "admin";
+
+    private static final String DEFAULT_PASSWORD = "quest";
+
+    private static final Integer POSTGRES_PORT = 8812;
+
+    private static final Integer REST_PORT = 9000;
+
+    private static final Integer ILP_PORT = 9009;
+
+    static final String TEST_QUERY = "SELECT 1";
+
+    static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("questdb/questdb");
+
+    public QuestDBContainer(@NonNull String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public QuestDBContainer(DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        withExposedPorts(POSTGRES_PORT, REST_PORT, ILP_PORT);
+        withCommitLag(DEFAULT_COMMIT_LAG_MS);
+        waitingFor(Wait.forLogMessage("(?i).*A server-main enjoy.*", 1));
+    }
+
+    public QuestDBContainer withCommitLag(int commitLagMillis) {
+        addEnv("QDB_CAIRO_COMMIT_LAG", String.valueOf(commitLagMillis));
+        return this;
+    }
```

Comment:
I think we don't need this method in the API. Instead, I would prefer documenting the usage of 
```
new QuestDBContainer().withEnv("QDB_CAIRO_COMMIT_LAG", 4711)
```